sklearn/utils/weight_vector.pyx

```diff
@@ -197,3 +203,28 @@ cdef class WeightVector(object):
     cdef double norm(self) nogil:
         """The L2 norm of the weight vector. """
         return sqrt(self.sq_norm)
+
+    cdef double get_max_weight(self) nogil:
+        """Computes the maximum absolute value over all features"""
+        cdef double max_weight = 0.0
+        cdef int n_features = self.n_features
+        cdef double* w_data_ptr = self.w_data_ptr
+
+        for j in range(n_features):
+            max_weight = fmax(max_weight, fabs(w_data_ptr[j]))
+        return max_weight
+
+    cdef double get_max_change(self, double **previous_weights) nogil:
```

Comment:
I don't remember why we didn't use `weight_vectors` for SAG in the first place, but it will need a bit of work to update it.